pkg/informer/informer.go

```diff
@@ -58,14 +59,64 @@ const (
 	byGroupVersionResourceIndex = "byGroupVersionResource"
 )
 
-// DynamicDiscoverySharedInformerFactory is a SharedInformerFactory that
+// GVRPartialMetadata provides the required metadata about a GVR
+// for which an informer should be started.
+type GVRPartialMetadata struct {
+	Names apiextensionsv1.CustomResourceDefinitionNames
+	Scope apiextensionsv1.ResourceScope
+}
+
+// GVRSource is a source of information about available GVRs,
+// and provides a way to register for changes in the available GVRs (addition or removal)
+// so that the list of informers can be updated.
+type GVRSource interface {
+	// GVRs returns the required metadata about all GVRs known by the GVRSource
+	GVRs() map[schema.GroupVersionResource]GVRPartialMetadata
+
+	// Synced returns true if informers used by this GVRSource are synced
+	Synced() bool
+
+	// Changes returns a channel to which the GVRSource writes whenever
+	// its list of known GVRs has changed
+	Changes() <-chan struct{}
+
+	// FilterBuiltinGVRs is a function that allows filtering out
+	// builtin GVRs that should not be considered.
+	// The function should return false to ignore a builtin GVR,
+	// and true to keep it.
+	FilterBuiltinGVRs(gvr schema.GroupVersionResource) bool
```

Comment:
strange method. If something is not to be considered by the consumer of this interface, don't list it in `GVRs()`.

Reply:
It mainly came from refactoring the existing code. I removed it and included the addition of builtin GVRs in the CRD-based `GVRSource`.